### DISCUSSION TITLE
Logreg vs pmm for imputing dichotomous variables

### DISCUSSION THREAD

**Discussion by angelrodriguez2020** [amices/mice#383] [Missing data methodology]
Dear all,

I'm a little bit confused about conflicting recommendations on this issue.
The default method to impute a dichotomous variable in mice is logreg, what makes conceptual sense. But I've found the following advice: 'For sparse categorical data, it may be better to use method pmm instead of logreg, polr or polyreg" 
(https://stefvanbuuren.name/fimd/sec-modelform.html)'. I would consider a dichotomous variable a very sparse categorical variable. But in the mice paper of the Journal of Statistical Software (p42) it is said: 'For variable with many categories,
we therefore recommend pmm'. So apparently pmm is preferred over other methods in categorical variables with very few or many categories. 
Is this advice based on efficiency of the pmm method over logreg and polyreg, or are there more substantive issues there?

Many thanks.

**Comment by stefvanbuuren** [comment]
Methods `polr` and `polyreg` become slow and unstable if data are sparse, especially if the outcome categories have low prevalence or if there's a multitude of predictors. In those cases, we've found `pmm` to be a suitable alternative, not only because it is faster, but also because it is statistically better behaved (more robust). Hence the advice.

**Reply by hls-psych** [reply]
Hi, I have some categorical and binary data which is sparse. If I use pmm, can the class of my categorical and binary data be 'factor'/'ordered factor' or should I convert them to 'numeric' if using pmm? Thanks so much

**Reply by hanneoberman** [reply]
Converting the factor variables is not required. However, due to the stochastic nature of the imputation process, the imputations may differ from results obtained on numeric data:

``` r
library(mice)
# all numeric variables
dat_num <- nhanes
imp_num <- mice(
  dat_num,
  seed = 123,
  visitSequence = names(dat_num),
  print = FALSE
)
imp_num_long <- complete(imp_num, "long")
# with hypertension as categeorical variable
dat_cat <- nhanes
dat_cat$hyp <- as.factor(dat_cat$hyp)
imp_cat <- mice(
  dat_cat,
  method = "pmm",
  seed = 123,
  visitSequence = names(dat_num),
  print = FALSE
)
imp_cat_long <- complete(imp_cat, "long") |>
  dplyr::mutate(hyp = as.numeric(hyp))
# compare imputed data
all.equal(imp_num_long, imp_cat_long)
#> [1] "Component \"bmi\": Mean relative difference: 0.1905533"
#> [2] "Component \"hyp\": Mean relative difference: 0.6428571"
#> [3] "Component \"chl\": Mean relative difference: 0.2404072"
```

<sup>Created on 2024-11-08 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Reply by hls-psych** [reply]
Thank you

**Comment by stefvanbuuren** [comment]
See <https://journals.sagepub.com/doi/full/10.1177/09622802231198795> for a comparison.
